sdk/inc/azure/core/az_json.h

```diff
@@ -104,6 +104,36 @@ az_json_token_get_uint64(az_json_token const* json_token, uint64_t* out_value);
 AZ_NODISCARD az_result
 az_json_token_get_uint32(az_json_token const* json_token, uint32_t* out_value);
 
+/**
+ * @brief Returns the JSON token's number as a 64-bit signed integer.
+ *
+ * @param json_token A pointer to an #az_json_token instance.
+ * @param out_value A pointer to a variable to receive the value.
+ * @return AZ_OK if the number is returned.<br>
+ * AZ_ERROR_JSON_INVALID_STATE if the kind != AZ_JSON_TOKEN_NUMBER.
+ */
+AZ_NODISCARD az_result az_json_token_get_int64(az_json_token const* json_token, int64_t* out_value);
```

Comment:
I thought we weren't going to have any of these functions as the customer can just call atoX functions passing in the token's span.

Reply:
We could do that, but it forces the caller to do a check on token kind, which in practice (for example usages within the SDK), folks don't do. So, these are convenience APIs to make it easier for folks to get the value they are looking for without having to do an extra token kind check themselves.

https://github.com/Azure/azure-sdk-for-c/blob/24c2c9128f29ea99ad89db4132a019da1d2fabfb/sdk/src/azure/iot/az_iot_provisioning_client.c#L257-L264

Reply:
Either they assume the value is of a certain type and just call the parse function.
Or, they have to do the check anyway: if tokenKind == number, then parse the number.
Both parse methods (span & json) return az_result

Reply:
It is also odd to only have `get_boolean` and `get_string`, on `az_json_token` but not support the numeric types.

Reply:
>Both parse methods (span & json) return az_result

True, but one returns an error due to mismatch of the expected schema, and the other returns for invalid characters to parse for that token type.

For example, are we OK with something taking a JSON string token, not checking the token kind, and passing that to the span APIs and get back a number, **accidentally**? The parser is returning tokens, and I, as the caller, should be able to call the method of  the type I expect which should fail otherwise. (FYI, that scenario is possible today, but the user has to explicitly try to do that, by knowing to grab the span slice and pass that to the span atoi API).

{"name":"123"} // If we use the az_span API and pass in "123", it will succeed.

If we are fine with that, I can remove it. What do we do about the inconsistency with `get_boolean` and `get_string` though, which still live on `az_json_token`?

Reply:
Discussed offline, keeping these methods. They are convenience methods that are quite small in size so worth the trade off.